File: app/assets/javascripts/vendor.js

if ($('.vendor-show')) {
  $.getJSON( location.href +".json",function(data){
    $('#vendor-table').DataTable({
      data: data,
      columns: [
                { title: "City", className: ""},
                { title: "Product Name",  className: ""},
                { title: "Price", className: "" },
                { title: "Quantity", className: "" },
                { title: "Date and Time", className: "" }
              ]
    });
  })
}

$.getJSON( "/vendors.json",function(data){
  $('#vendor-index-table').DataTable({
    data: data,
    columns: [
              { title: "Vendor", className: ""}
            ]
  });

  $('#vendor-index-table').on('click', 'tbody td', function() {
    console.log(data[this.parentNode.rowIndex - 1])
    window.location.href = '/vendor/' + data[this.parentNode.rowIndex - 1]
  })


})

File: app/models/vendor.rb
class Vendor < ApplicationRecord
	has_many :orders

	def self.all_stores
		stores = []
		self.all.each do |s|
			stores.push([s.name])
		end
		stores
	end


	def store_page
		store_page = []
		orders.each do |o|
			store_page.push([o.city, o.name, o.price, o.quantity, o.date])
		end
		store_page
	end



end
